src/main/java/slimeknights/tconstruct/smeltery/tileentity/TileSearedFurnace.java

```diff
@@ -0,0 +1,562 @@
+  @Override
+  protected boolean onItemFinishedHeating(ItemStack stack, int slot) {
+    ItemStack result = FurnaceRecipes.instance().getSmeltingResult(stack);
+
+    if(result == null) {
+      return false;
+    }
+    
+    // rememeber, we can smelt a whole stack at once
+    result = result.copy();
+    result.stackSize *= stack.stackSize;
+    
+    setInventorySlotContents(slot, result);
+    // we set to 1 so we get positive infinity instead of NaN for progress, since NaN is already defined as no recipe
+    itemTemperatures[slot] = 1;
+    itemTempRequired[slot] = 0;
+
+    // we return false since the itemstack does not leave the slot, otherwise any data we set here is lost (which really would just be the temp)
+    return false;
+  }
+
+  @Override
+  protected void consumeFuel() {
+    // no need to consume fuel
+    if(hasFuel()) {
+      return;
+    }
+
+    // get current tank
+    searchForFuel();
+
+    // got a tank?
+    if(currentTank != null) {
+      // consume fuel!
+      TileEntity te = worldObj.getTileEntity(currentTank);
+      if(te instanceof TileTank) {
+        IFluidTank tank = ((TileTank) te).getInternalTank();
+
+        FluidStack liquid = tank.getFluid();
+        if(liquid != null) {
+          FluidStack in = liquid.copy();
+          int bonusFuel = TinkerRegistry.consumeSmelteryFuel(in);
+          int amount = liquid.amount - in.amount;
+          FluidStack drained = tank.drain(amount, false);
+
+          // we can drain. actually drain and add the fuel
+          if(drained != null && drained.amount == amount) {
+            tank.drain(amount, true);
+            fuelQuality = bonusFuel;
+            currentFuel = drained.copy();
+            addFuel(bonusFuel, drained.getFluid().getTemperature(drained) - 300); // convert to degree celcius
+
+            // notify client of fuel/temperature changes
+            if(worldObj != null && !worldObj.isRemote) {
+              TinkerNetwork.sendToAll(new SmelteryFuelUpdatePacket(pos, currentTank, temperature, currentFuel));
+            }
+            
+            return;
+          }
+        }
+        
+        fuelQuality = 0;
+      }
+    }
+  }
+
+  /**
+   * Locates a tank containing fuel, if one exists
+   * @return true if successful
+   */
+  private void searchForFuel() {
+    // is the current tank still up to date?
+    if(currentTank != null && hasFuel(currentTank, currentFuel)) {
+      return;
+    }
+
+    // nope, current tank is empty, check others for same fuel
+    for(BlockPos pos : tanks) {
+      if(hasFuel(pos, currentFuel)) {
+        currentTank = pos;
+        return;
+      }
+    }
+
+    // nothing found, try again with new fuel
+    for(BlockPos pos : tanks) {
+      if(hasFuel(pos, null)) {
+        currentTank = pos;
+        return;
+      }
+    }
+
+    currentTank = null;
+  }
+
+  // checks if the given location has a fluid tank that contains fuel
+  private boolean hasFuel(BlockPos pos, FluidStack preference) {
```

Comment:
The packet renaming is in reference to the one from `consumeFuel`, as now the packet refers to anything using TileTankFuelHeatingStructure instead of just smelteries.